### PR TITLE
feat: add dev guide about and quick start sections.

### PR DESCRIPTION
**Pull request by Peefy**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

In #61, the directory of KCL developing guide is discussed and designed, this PR is to fill in the content of the first two chapters (About the guide & Quick Start), and other content is written in another PR.

#### 2. What is the scope of this PR (e.g. component or file name):

/docs/dev_guide/1.about_this_guide.md
/docs/dev_guide/2.quick_start.md

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

Documentation improvements.

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by chai2010 — DISMISSED** [review]
LGTM

**Review by zong-zhe — DISMISSED** [review]
LGTM

File: docs/dev_guide/2.quick_start.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+- `git`
+- `Rust 1.60+`
+- `Python 3.7+`
+- `LLVM 12`
+
+Please add the LLVM installation location to `LLVM_SYS_120_PREFIX` and the `$PATH`.
+
+#### Docker
+
+Use the image `kusionstack/kclvm-builder`
```
Comment (ldxdl) [inline docs/dev_guide/2.quick_start.md:103]: + Change the desc with a runnable cli
+ Make the docker-env as the recommended method

File: docs/dev_guide/2.quick_start.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+### Hardware
```
Comment (ldxdl) [inline docs/dev_guide/2.quick_start.md:105]: You'd better put dev environment information at the beginning of the paragraph

File: docs/dev_guide/2.quick_start.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+In the top level of the repo:
+
+```
+./run.sh -a test
+```
+
+See the chapters on building and testing for more details.
+
+## Contributor Procedures
+
```
Comment (ldxdl) [inline docs/dev_guide/2.quick_start.md:141]: Consider better organizing the content, such as:

### Create an Issue

### Create a PR

#### Bug Fixes or "Normal" Code Changes

#### New Features

File: docs/dev_guide/2.quick_start.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+#### macOS and OS X
+
+- `git`
+- `Python 3.7+`
+- `Rust 1.60+`
+- `openssl@1.1`
+- `LLVM 12`
+
+You can install `openssl@1.1` using `brew`:
```
Comment (ldxdl) [inline docs/dev_guide/2.quick_start.md:30]: remove openssl install cause build-in py is removed.

Comment (ldxdl) [inline docs/dev_guide/2.quick_start.md:30]: remove openssl install cause build-in py is removed.

File: docs/dev_guide/2.quick_start.md (hunk 1 above)
Comment (Peefy) [reply docs/dev_guide/2.quick_start.md]: > * Change the desc with a runnable cli
> * Make the docker-env as the recommended method

Modified, PTAL

File: docs/dev_guide/2.quick_start.md (hunk 2 above)
Comment (Peefy) [reply docs/dev_guide/2.quick_start.md]: Modified, PTAL

File: docs/dev_guide/2.quick_start.md (hunk 3 above)
Comment (Peefy) [reply docs/dev_guide/2.quick_start.md:142]: > Consider better organizing the content, such as:
> 
> ### Create an Issue
> ### Create a PR
> #### Bug Fixes or "Normal" Code Changes
> #### New Features

Good suggestion, the content has been reorganized

File: docs/dev_guide/2.quick_start.md (hunk 4 above)
Comment (Peefy) [reply docs/dev_guide/2.quick_start.md]: > remove openssl install cause build-in py is removed.

removed

Comment (Peefy) [reply docs/dev_guide/2.quick_start.md]: > remove openssl install cause build-in py is removed.

removed

**Comment by ldxdl** [comment]
LGTM

**Review by ldxdl — APPROVED** [review]
LGTM

**Review by ldxdl — APPROVED** [review]

File: docs/dev_guide/2.quick_start.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+See the chapters on building and testing for more details.
+
+## Contributor Procedures
+
+### Create an Issue
+
+Every change should be accompanied by a dedicated tracking issue for that change. The main text of this issue should describe the change being made, with a focus on what users must do to fix their code. The issue should be approachable and practical; it may make sense to direct users to some other issue for the full details. The issue also serves as a place where users can comment with questions or other concerns.
+
+When you open a issue on the `KusionStack/KCLVM` repo, you need to to choose an issue template on this [page](https://github.com/KusionStack/KCLVM/issues/new/choose), you can choose a template according to different situations and fill in the corresponding content, and you also need to select appropriate labels for your issue to help classify and identify.
```
Comment (chai2010) [inline docs/dev_guide/2.quick_start.md:147]: open **an** issue

Comment (Peefy) [reply docs/dev_guide/2.quick_start.md]: > open **an** issue

fixed.

**Pull request merged**